libs/store_helper/kernel/ebpf_registry_helper.c

```diff
@@ -108,6 +109,40 @@ ebpf_create_registry_key(
     return _EBPF_RESULT(ZwCreateKey(key, KEY_WRITE, &object_attributes, 0, NULL, REG_OPTION_NON_VOLATILE, NULL));
 }
 
+_Must_inspect_result_ ebpf_result_t
+ebpf_open_registry_key(
+    ebpf_store_key_t root_key, _In_z_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_store_key_t* key)
+{
+    UNICODE_STRING registry_path;
+    OBJECT_ATTRIBUTES object_attributes = {0};
+
+    UNREFERENCED_PARAMETER(flags);
+
+    RtlInitUnicodeString(&registry_path, sub_key);
+    InitializeObjectAttributes(
+        &object_attributes, &registry_path, OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE, root_key, NULL);
+
+    return _EBPF_RESULT(ZwOpenKey(key, KEY_WRITE, &object_attributes));
+}
+
+_Must_inspect_result_ ebpf_result_t
+ebpf_delete_registry_tree(ebpf_store_key_t root_key, _In_opt_z_ const wchar_t* sub_key)
+{
+    UNREFERENCED_PARAMETER(root_key);
+    UNREFERENCED_PARAMETER(sub_key);
+
+    return EBPF_OPERATION_NOT_SUPPORTED;
+}
+
+_Must_inspect_result_ ebpf_result_t
+ebpf_delete_registry_tree_ansi(ebpf_store_key_t root_key, _In_opt_z_ const char* sub_key)
```

Comment:
Why is this API here?  I don't think we should have _ansi_ apis.

Reply:
I added an ansi version as in some cases, the caller does not have wide char string. Also, since MSDN has both the `A` and `W` versions of the API, I added 2 versions here also.

Reply:
Yea but this is a new code, and we can probably only have unicde friendly API?

Reply:
Right.  Please point me to an ebpf case where the caller cannot get a wide char string.  Otherwise, please delete this code as MSDN `A` apis are considered deprecated.

Reply:
Fixed.